File: backend/app/schemas/nursing.py
from datetime import date

from pydantic import BaseModel, Field

from app.models.nursing import AlertSeverity, BedStatus, CheckInStatus, Gender


class ElderCreate(BaseModel):
    name: str = Field(min_length=1, max_length=64)
    gender: Gender = Gender.unknown
    birthday: date | None = None
    phone: str | None = Field(default=None, max_length=32)
    family_contact: str | None = Field(default=None, max_length=64)
    health_summary: str | None = None


class RoomCreate(BaseModel):
    floor: str = Field(min_length=1, max_length=32)
    room_no: str = Field(min_length=1, max_length=32)
    room_type: str = Field(default="standard", max_length=64)


class BedCreate(BaseModel):
    room_id: int
    bed_no: str = Field(min_length=1, max_length=32)
    status: BedStatus = BedStatus.available


class NursingProjectCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    category: str = Field(default="daily", max_length=64)
    description: str | None = None
    price: float = 0


class AlertRecordCreate(BaseModel):
    elder_id: int | None = None
    device_name: str = Field(min_length=1, max_length=100)
    severity: AlertSeverity = AlertSeverity.medium
    content: str = Field(min_length=1)


class CheckInApplicationCreate(BaseModel):
    elder_id: int
    preferred_room_type: str | None = Field(default=None, max_length=64)
    care_needs: str | None = None
    status: CheckInStatus = CheckInStatus.draft

File: frontend/src/api/nursing.ts
export interface Elder {
  id: number;
  name: string;
  gender: string;
  phone?: string;
  family_contact?: string;
  health_summary?: string;
}

export interface Room {
  id: number;
  floor: string;
  room_no: string;
  room_type: string;
}

export interface Bed {
  id: number;
  room_id: number;
  bed_no: string;
  status: string;
}

export interface NursingProject {
  id: number;
  name: string;
  category: string;
  price: number;
}

export interface AlertRecord {
  id: number;
  device_name: string;
  severity: string;
  content: string;
  handled: boolean;
}

export async function listElders(): Promise<Elder[]> {
  return fetchJson('/api/nursing/elders');
}

export async function listRooms(): Promise<Room[]> {
  return fetchJson('/api/nursing/rooms');
}

export async function listBeds(): Promise<Bed[]> {
  return fetchJson('/api/nursing/beds');
}

export async function listNursingProjects(): Promise<NursingProject[]> {
  return fetchJson('/api/nursing/projects');
}

export async function listAlerts(): Promise<AlertRecord[]> {
  return fetchJson('/api/nursing/alerts');
}

async function fetchJson<T>(url: string): Promise<T> {
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('业务数据加载失败');
  }
  return response.json();
}
